### ISSUE TITLE
MAC not reported as online

### ISSUE THREAD

**Issue by gitbock** [gitbock/fritzboxtr064#1]
_Issue_:
- WebUI shows MAC as online
- Binding shows MAC as offline
- Other MACs are reported correctly as online/offline

_Log file_

```
2015-10-03 20:12:52.229 [DEBUG] [o.o.b.f.i.FritzboxTr064Binding] - FritzboxTr064 executing...
2015-10-03 20:12:53.046 [DEBUG] [o.o.b.f.i.FritzboxTr064Binding] - Parsing fbox response for maconline
2015-10-03 20:12:53.050 [DEBUG] [o.o.b.f.i.FritzboxTr064Binding] - parsed as 1
2015-10-03 20:12:53.075 [DEBUG] [o.o.b.f.i.FritzboxTr064Binding] - Parsing fbox response for maconline
2015-10-03 20:12:53.080 [DEBUG] [o.o.b.f.i.FritzboxTr064Binding] - MAC not known to FritzBox!
```

_References_
[pull request openhab](https://github.com/openhab/openhab/pull/2494#issuecomment-145274238)

_versions affected_
- v0.1
- v0.2

_Environment_
- All addressing is done using DHCP server of Fritzbox

**Comment by gitbock** [comment]
This is a known bug. Sometimes the fritzbox is not reporting the proper state. You may try one of the following resolutions:

### Check

Check if the problem is because of the fritzbox bug or the binding itself. Download the [perl script](https://github.com/gitbock/fritzboxtr064/blob/master/lib/fb-mac-online.pl), adapt the header part in the script so that it matches for MAC and fritzbox ip. Execute the script and watch the output. If the state is reported properly, it's  a binding problem -> rise a new issue including debug logs and your configuration. If the state is reported incorrect, you may try the following options.

### Fritzbox DHCP reset
- enable debug logging as decribed in the [Readme file](https://github.com/gitbock/fritzboxtr064/blob/master/README.md)
- configure the device in question to use static addressing, not using DHCP of FritzBox
- delete the device from FritzBox Netzwerkübersicht
- let the binding execute and look if behavior is changed
- re-enable DHCP on the device, so FritzBox assigns an IP address again
- let the binding execute and look if behavior is changed
- watch the openhab.log for debug messages; especially those starting with "Raw SOAP Request/Response ..."

### Fritzbox MAC reservation
- set/unset the MAC reservation for the device in fbox
- let the binding execute and look if behavior is changed

**Comment by thaake** [comment]
I tried all above with no effect.
Do I have to restart openhab to have changes in logbook.xml take effect?
That I didn't do today because I don't wanted to risk an unstable openhab before leaving home ;-)

**Comment by gitbock** [comment]
@thaake
No, I think you do not need to restart. It only should take a while until the settings of logback.xml get active. At the moment I do not have another idea, sorry.

If you are using a Linux based system, you may also try [my perl Skript](https://www.bock-systems.de/?p=161), which basically does the same, but not using openHAB. Just to get another idea...

**Comment by thaake** [comment]
@gitbock 
I'm definitely running v0.2 and have the correct debug entries in logback.xml, but I don't get any SOAP entry in openhab.log.
Behavior of binding is still weird. Rest of openhab is fine.

**Comment by thaake** [comment]
Seems to need a restart to take effect for the entries in logback.xml.

I just gave it a try, but with your debug entry openhab doesn't start correct. It didn't write an openhab.log anyway.
It deleted the debug entries and restarted. all is fine again.
Something seems to be very wrong with debug enabled in the v0.2.

**Comment by thaake** [comment]
Anything new?

**Comment by gitbock** [comment]
Don't have an idea at the moment.
Did you try the perl script as I suggested in [this post](https://github.com/gitbock/fritzboxtr064/issues/1#issuecomment-145575914)?

**Comment by thaake** [comment]
@gitbock 

I tried the perl script, but I didn't manage to get it running.
My system is a synology and I got a lot of trouble with the install of perl and the required packages.
I gave up in the end. Sorry.

**Comment by gitbock** [comment]
@thaake 
If you are still interested: Some people reported that the binding is working as soon as they upgraded to java 1.8.0. You may have a look at #3 or #4

**Comment by thaake** [comment]
@gitbock 

Thanks for all your work. I followed all those other "threads" and tried it again: It works for me, too.

I did two things, unfortunately at once, FB Laborversion (after 6.30 didn't do the trick) and Java 8.
After it runs like a charm. I'm testing all the details, but did not come up with any problem till now.

I think Java 8 is the game changer here. It's a bit tricky the get Java 8 running on Sinology, because of the Java manager, it only supports up to J7. But it you rename the J8 file to a 7, the manager does not get it and installs J8. Thats it.

I'll keep you posted about my testing. Till now, all thumbs up!!!
Thanks again.

**Comment by gitbock** [comment]
@thaake 
Thanks for your valuable feedback, which I appreciate much!
You are welcome :)

**Comment by hjw2015** [comment]
@gitbock 

Fiddling around with this MAC online test the last two days - here are my findings:
- if a device get's renamed in the Web UI it is no longer visible using the perl script (or the binding)
- deleting the device and reconnecting it to the FB WLAN makes it again visible
  -- there may pop-up a "reset" button in the device properties - using this button sometimes seems to be enough to make a device visible

Long story short: devices are gone after renaming them in the Web UI in my scenario

Context:
- FB 7390, FW 6.30
- DHCP server disabled (provided by raspberry)
- TR064 binding 0.7

**Comment by gitbock** [comment]
@hjw2015 
Thanks for the tests. I was not aware that such a reset button can appear :)

**Comment by hjw2015** [comment]
In my case it shows up only if I change the name of a device. Alternatively
you can delete devices when you restrict access to only known MAC addresses
in the WLAN view.

gitbock notifications@github.com schrieb am Sa., 2. Jan. 2016 13:51:

**Comment by GeneralLuzi** [comment]
Hi,

Summary:
It depends on the name of device given at the fritz.box UI. If it is on the last position of the list, i won't work, where as at the beginning to the second last position is is fine.

Long version:
this issue is still not closed for me. I have the latest SDK /  JRE on me raspberry 2 and the presence detection does not work for 1 device. 

I tested this issue and my observation is the following:
If the device is named in that way, that it is located on the last position of the networklist then the present detection won't work. 

See the following screenshot:
![error_fritzbox](https://cloud.githubusercontent.com/assets/10353223/15292814/41beea14-1b85-11e6-91bb-c0f065e507e1.jpg)
The device starting with ZZ is not available.

Can someone check if I'm right please? If so, then someone should report this to AVM.

BR,
GeneralLuzi

**Comment by hjw2015** [comment]
Problem seems to be fixed with latest update of FritzBox to 6.51 for me.
Worked flawless the last week.

BR,
HJW

GeneralLuzi notifications@github.com schrieb am Mo., 16. Mai 2016, 16:45:

**Comment by GeneralLuzi** [comment]
PS.: I'm using the FRITZ!OS 06.06. Perhaps this issue does not appear in newer versions...

**Comment by GeneralLuzi** [comment]
7 minutes to late ;) thanks for the reply

Then this should be the workaround for anyone who is using an older fritzbox...

this thread can be closed now...
